six/modules/c++/cphd/include/cphd/CPHDXMLControl.h

```diff
@@ -46,75 +54,135 @@ class CPHDXMLControl : public six::XMLParser
     //!  Constructor
     CPHDXMLControl();
 
-    CPHDXMLControl(logging::Logger* log, bool ownLog = false);
+    //! Constructor
+    CPHDXMLControl(logging::Logger* log, bool ownLog);
 
-    /*!
-     *  This function takes in a Metadata object and converts
-     *  it to a new-allocated XML DOM.
-     */
-    std::auto_ptr<xml::lite::Document> toXML(const Metadata& metadata);
+    //! Constructor
+    CPHDXMLControl(logging::Logger* log, bool ownLog, std::vector<std::string>& schemaPaths);
+
+    // To string
+    std::string toXMLString(const Metadata& metadata);
 
-    /*!
-     *  Function takes a DOM Document* node and creates a new-allocated
-     *  CPHDData* populated by the DOM.
-     */
     std::auto_ptr<Metadata> fromXML(const xml::lite::Document* doc);
 
+    //! (Testing) Parses specifc XML blocks specified by nodeNames vector
+    std::auto_ptr<Metadata> fromXML(const xml::lite::Document* doc, std::vector<std::string>& nodeNames);
+
     std::auto_ptr<Metadata> fromXML(const std::string& xmlString);
 
-    std::string toXMLString(const Metadata& metadata);
-    size_t getXMLsize(const Metadata& metadata);
+    //setter
+    void setSchemaPaths(std::vector<std::string> schemaPaths)
```

Comment:
I'd rather not expose these unless there's some pressing need. If a caller wants to set the schema paths, they can just do that in the constructor